app/javascript/flavours/polyam/features/notifications_v2/components/notification_group_with_status.tsx

```diff
@@ -90,6 +91,39 @@ export const NotificationGroupWithStatus: React.FC<{
     [dispatch, statusId],
   );
 
+  // Polyam: collapsing
+
+  const collapseEnabled = useAppSelector(
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access
+    (state) => state.local_settings.getIn(['collapsed', 'enabled']) as boolean,
+  );
+
+  const autoCollapse = useAppSelector(
+    (state) =>
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+      (state.local_settings
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        .getIn(['collapsed', 'auto'])
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        .get('all') as boolean) ||
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+      (state.local_settings
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        .getIn(['collapsed', 'auto'])
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        .get('notifications') as boolean),
+  );
```

Comment:
This is extremely awkward, but not much can be done until the state is correctly typed, which will take a while.

Reply:
Well, I tried to make it less awkward but prettier had other ideas

Reply:
Typecasting gets rid of the comment cascade, but due to the formatting, it's not exactly more readable.